client-js/app/elements/labrad-plot1d.ts

```diff
@@ -252,4 +258,46 @@ export class Plot1D extends polymer.Base implements Plot {
          .attr('d', this.line);
     }
   }
+  
+  private reset() {
+    this.xScale.domain([this.limits.xMin, this.limits.xMax]);
+    this.yScale.domain([this.limits.yMin, this.limits.yMax]);
+    this.xAxis.scale(this.xScale);
+    this.yAxis.scale(this.yScale);
+    this.zoom.x(this.xScale);
+    this.zoom.y(this.yScale);
+    this.handleZoom();
+  }
+
+  private zoomRectangle() {
+    var e = this,
+        origin = d3.mouse(e),
+        rect = e.svg.append('rect')
+	    .attr('stroke', 'red')
+            .attr('fill-opacity', 0.5);
+
+    e.svg
+        .on('mousemove.zoomRect', function() {
+          var m = d3.mouse(e);
```

Comment:
Here, too, I think it would be clearer to just pull out the separate components:
```javascript
var [x, y] = d3.mouse(e);
```